test/handlers/fee/withOracle/calculateFee.js

```diff
@@ -0,0 +1,205 @@
+        assert.equal(Ethers.utils.formatEther(res.fee.toString()), "0.00491802");
+        assert.equal(res.tokenAddress, ERC20MintableInstance.address);
+    });
+
+    it("should return percent fee", async () => {
+        const tokenAmount = Ethers.utils.parseEther("1");
+        const depositData = Helpers.createERCDepositData(tokenAmount, 20, recipientAddress);  
+        const oracleResponse = {
+            ber: Ethers.utils.parseEther("0.000533"),
+            ter: Ethers.utils.parseEther("1.63934"),
+            dstGasPrice: Ethers.utils.parseUnits("30000000000", "wei"),
+            timestamp: Math.floor(new Date().valueOf() / 1000),
+            fromDomainID: domainID,
+            toDomainID: domainID,
+            resourceID
+        };
+
+        const feeData = Helpers.createOracleFeeData(oracleResponse, oracle.privateKey, tokenAmount);
+        const res = await FeeHandlerWithOracleInstance.calculateFee.call(sender, domainID, domainID, resourceID, depositData, feeData);
+        assert.equal(web3.utils.fromWei(res.fee, "ether"), "0.05");
+        assert.equal(res.tokenAddress, ERC20MintableInstance.address);
+    });
+
+    it("should return fee to cover tx cost if percent fee does not cover tx cost", async () => {
+        const tokenAmount = 100;      
+        const depositData = Helpers.createERCDepositData(tokenAmount, 20, recipientAddress);  
+
+        const oracleResponse = {
+            ber: Ethers.utils.parseEther("0.0005"),
+            ter: Ethers.utils.parseEther("1.5"),
+            dstGasPrice: Ethers.utils.parseUnits("30000000000", "wei"),
+            timestamp: Math.floor(new Date().valueOf() / 1000),
+            fromDomainID: domainID,
+            toDomainID: domainID,
+            resourceID
+        };
+
+        const feeData = Helpers.createOracleFeeData(oracleResponse, oracle.privateKey, tokenAmount);
+        const res = await FeeHandlerWithOracleInstance.calculateFee.call(sender, domainID, domainID, resourceID, depositData, feeData);
+        assert.equal(Ethers.utils.formatEther(res.fee.toString()), "0.0045");
+        assert.equal(res.tokenAddress, ERC20MintableInstance.address);
+    });
+
+    it("should not calculate fee if fee data is misformed", async () => {
+        const tokenAmount = 100;      
+        const depositData = Helpers.createERCDepositData(tokenAmount, 20, recipientAddress);  
+
+        const oracleResponse = {
+            ber: Ethers.utils.parseEther("0.0005"),
+            ter: Ethers.utils.parseEther("1.5"),
+            dstGasPrice: Ethers.utils.parseUnits("30000000000", "wei"),
+            timestamp: Math.floor(new Date().valueOf() / 1000),
+            fromDomainID: domainID,
+            toDomainID: domainID,
+            resourceID
+        };
+
+        const feeData = Helpers.createOracleFeeData(oracleResponse, oracle.privateKey, tokenAmount) + "1";
```

Comment:
Not valid bytearray, + "1" adds only 4 bits to the data, which is incorrect and probably treated by web3 no as bytearray at all.